### ISSUE TITLE
Including a file in multiple targets

### ISSUE THREAD

**Issue by divergio** [jcampbell05/xcake#119]
I have a project that uses multiple targets inside one project. Each target is a different version of the app, i.e. a target for `CakeMania`, a target for `CakeMania for Kids`. I believe this is a fairly standard way of managing app variants. 

Most of the files are shared between the targets, but some targets include additional SDKs and of course some different files (for icons, built-in assets, configuration, etc).

CocoaPods helps us organize our 3rd party components using Ruby functions, making the configuration very clean and easy to read. I hoped Xcake could do the same for our project files. 

The Cakefile looks like this (omitting project configuration block): 
```
def cakemania_shared(target) 
  target.include_files = ["Source/App/APIHelper.*"]
end
    
application_for :ios, 8.0 do |target|
    target.name = "CakeMania"
    cakemania_shared(target)
  end
  
application_for :ios, 8.0 do |target|
      target.name = "CakeMania for Kids"
      cakemania_shared(target)
end
```

The desired output project file would have one copy of the APIHelper.m file, with membership in both targets. 
<img width="366" alt="Desired output" src="https://cloud.githubusercontent.com/assets/359631/20095416/5f546a50-a5e0-11e6-9c47-0d54740c7134.png">

The current output project file includes the file in the project twice, with non-overlapping membership. 
<img width="400" alt="Current output" src="https://cloud.githubusercontent.com/assets/359631/20095436/6f78997e-a5e0-11e6-9b4c-3ce0e73a901f.png">

**Comment by jcampbell05** [comment]
Ah nots great, Xcake should definatly be doing the first screenshot. I'll try and recreate this locally and issue a fix.

**Comment by maximkhatskevich** [comment]
Agree this is important and should be a priority.

**Comment by jcampbell05** [comment]
:) Going to look at this this weekend and try to fix all the XC 8 issues

**Comment by eiditakahashi** [comment]
Hey guys! News on this issue?

For now I'm using version 0.6.25 since multiplying files don't happen.

**Comment by jcampbell05** [comment]
@eiditakahashi  I've done some work but I'm currently unit testing it. That would make sense there was a change in the architecture in 0.7 so maybe that broke it.

could you upload an example project with the issue ? I would love to test it against your project.

**Comment by eiditakahashi** [comment]
@jcampbell05 here it is:
test project running 0.7.1 https://www.dropbox.com/s/4i0h4jv580mps7c/test.zip?dl=0

**Comment by eiditakahashi** [comment]
@jcampbell05 yeah, it works! Not duplicating anymore! But now I am facing another problem related to `build_phase` I will open another issue for this. Thanks!
